tests/components/tedee/test_lock.py

```diff
@@ -0,0 +1,243 @@
+
+    state = hass.states.get("lock.lock_2c3d")
+    assert state
+    assert state.attributes.get(ATTR_DEVICE_CLASS) is None
+    assert state.attributes.get(ATTR_FRIENDLY_NAME) == "Lock-2C3D"
+    assert state.state == STATE_UNLOCKED
+
+    entry = entity_registry.async_get(state.entity_id)
+    assert entry
+    assert entry.device_id
+    assert entry.unique_id == "98765-lock"
+
+    device = device_registry.async_get(entry.device_id)
+    assert device
+    assert device.configuration_url is None
+    assert device.entry_type is None
+    assert device.hw_version is None
+    assert device.identifiers == {(DOMAIN, "98765")}
+    assert device.manufacturer == "Tedee"
+    assert device.name == "Lock-2C3D"
+    assert device.model == "Tedee GO"
+
+    with pytest.raises(
+        HomeAssistantError,
+        match="Entity lock.lock_2c3d does not support this service.",
+    ):
+        await hass.services.async_call(
+            LOCK_DOMAIN,
+            SERVICE_OPEN,
+            {
+                ATTR_ENTITY_ID: "lock.lock_2c3d",
+            },
+            blocking=True,
+        )
+
+    assert len(mock_tedee.open.mock_calls) == 0
+
+
+@pytest.mark.parametrize(
+    ("side_effect", "result_exception", "reason"),
+    [
+        (
+            TedeeClientException("Boom."),
+            UpdateFailed,
+            "Querying API failed. Error: Boom.",
+        ),
+        (
+            TedeeLocalAuthException(""),
+            ConfigEntryError,
+            "Authentication failed. Local access token is invalid",
+        ),
+        (
+            TedeeDataUpdateException("Boom."),
+            UpdateFailed,
+            "Error while updating data: Boom.",
+        ),
+        (TimeoutError("Boom."), UpdateFailed, "Querying API failed. Error: Boom."),
+    ],
+)
+async def test_coordinator_data_update_failures(
+    hass: HomeAssistant,
+    mock_tedee: MagicMock,
+    mock_config_entry: MockConfigEntry,
+    side_effect: Exception,
+    result_exception: type,
+    reason: str,
+) -> None:
+    """Test coordinator data update fails."""
+    mock_tedee.sync.side_effect = side_effect
+    coordinator = hass.data[DOMAIN][mock_config_entry.entry_id]
+
+    with pytest.raises(result_exception) as exc:
+        await coordinator._async_update_data()
+    assert str(exc.value) == reason
+
+    mock_tedee.get_locks.side_effect = side_effect
+    with pytest.raises(result_exception) as exc:
+        await coordinator._async_update_data()
+    assert str(exc.value) == reason
+
+
+async def test_coordinator_no_locks(
+    hass: HomeAssistant,
+    mock_tedee: MagicMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test coordinator no locks."""
+    mock_tedee.locks_dict = {}
+    coordinator = hass.data[DOMAIN][mock_config_entry.entry_id]
+    with pytest.raises(UpdateFailed) as exc:
+        await coordinator._async_update_data()
+    assert str(exc.value) == "No locks found in your account"
```

Comment:
Please don't test the itnernals, but test the effect what these issues have on the entities

Reply:
would you have an example for that?